controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -267,7 +268,7 @@ public CompletableFuture<CreateStreamStatus.Status> createRGStream(String scope,
                         log.debug(requestId, "Creating StateSynchronizer Stream {}", stream);
                         return createStreamRetryOnLockFailure(scope,
                                 stream,
-                                config,
+                                config.toBuilder().retentionPolicy(retentionPolicy).build(),
```

Comment:
This solves the issue for reader groups. However it does not solve it for State synchronizer streams or RevisionedStreamClients.